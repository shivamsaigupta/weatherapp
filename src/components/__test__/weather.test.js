import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Weather from "../weather.component";
import { faSun, faCloudRain } from "@fortawesome/free-solid-svg-icons";

import "@testing-library/jest-dom";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Weather />, div);
});

it("renders weather details correctly", () => {
  const weather = shallow(
    <Weather
      location="Mumbai"
      temp="234"
      precip="Sunny"
      loading={false}
      icon={faSun}
    ></Weather>
  );
  expect(weather).toMatchSnapshot();
});

it("renders weather details correctly 2", () => {
  const weather = shallow(
    <Weather
      location="Delhi"
      temp="164"
      loading={false}
      precip="Rain"
      icon={faCloudRain}
    ></Weather>
  );
  expect(weather).toMatchSnapshot();
});

it("renders if no props are passed", () => {
  const weather = shallow(<Weather />);
  expect(weather).toMatchSnapshot();
});
