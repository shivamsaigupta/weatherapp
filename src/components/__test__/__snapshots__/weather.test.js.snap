// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders if no props are passed 1`] = `
<div
  className="container"
  style={
    Object {
      "marginTop": 3,
      "opacity": 0.8,
    }
  }
>
  <div
    className="card py-4"
  >
    <div>
      <h1
        data-testid="location"
      />
      <h5
        className="py-4"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={null}
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="6x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </h5>
      <h1
        className="py-2"
        data-testid="temp"
      >
        °
      </h1>
      <h4
        className="py-3"
        data-testid="precip"
      />
    </div>
  </div>
</div>
`;

exports[`renders weather details correctly 1`] = `
<div
  className="container"
  style={
    Object {
      "marginTop": 3,
      "opacity": 0.8,
    }
  }
>
  <div
    className="card py-4"
  >
    <div>
      <h1
        data-testid="location"
      >
        Mumbai
      </h1>
      <h5
        className="py-4"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f185",
                "M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z",
              ],
              "iconName": "sun",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="6x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </h5>
      <h1
        className="py-2"
        data-testid="temp"
      >
        234
        °
      </h1>
      <h4
        className="py-3"
        data-testid="precip"
      >
        Sunny
      </h4>
    </div>
  </div>
</div>
`;

exports[`renders weather details correctly 2 1`] = `
<div
  className="container"
  style={
    Object {
      "marginTop": 3,
      "opacity": 0.8,
    }
  }
>
  <div
    className="card py-4"
  >
    <div>
      <h1
        data-testid="location"
      >
        Delhi
      </h1>
      <h5
        className="py-4"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f73d",
                "M416 128c-.6 0-1.1.2-1.6.2 1.1-5.2 1.6-10.6 1.6-16.2 0-44.2-35.8-80-80-80-24.6 0-46.3 11.3-61 28.8C256.4 24.8 219.3 0 176 0 114.1 0 64 50.1 64 112c0 7.3.8 14.3 2.1 21.2C27.8 145.8 0 181.5 0 224c0 53 43 96 96 96h320c53 0 96-43 96-96s-43-96-96-96zM88 374.2c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0zm160 0c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0zm160 0c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0z",
              ],
              "iconName": "cloud-rain",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="6x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </h5>
      <h1
        className="py-2"
        data-testid="temp"
      >
        164
        °
      </h1>
      <h4
        className="py-3"
        data-testid="precip"
      >
        Rain
      </h4>
    </div>
  </div>
</div>
`;
